src/Storages/MergeTree/IMergeTreeDataPart.h

```diff
@@ -65,6 +65,7 @@ enum class DataPartRemovalState : uint8_t
 class IMergeTreeDataPart : public std::enable_shared_from_this<IMergeTreeDataPart>, public DataPartStorageHolder
 {
 public:
+    void loadIndexWithLock() const { std::scoped_lock lock(index_mutex); loadIndex(); }
```

Comment:
This is a strange place for the function to be. Please move l. 68 to some better location.

Reply:
Agree, let's move it after the `using`s and the constructor, where the rest of the methods live.

Reply:
FIxed